runtime/src/main/java/org/corfudb/runtime/view/SequencerView.java

```diff
@@ -31,17 +35,52 @@ public SequencerView(CorfuRuntime runtime) {
      * @return The first token retrieved.
      */
     public TokenResponse nextToken(Set<UUID> streamIDs, int numTokens) {
-        return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
-                .nextToken(streamIDs, numTokens)));
+        if (numTokens == 0) {
+            Long ts = routeCall(streamIDs);
+            if (ts != null) {
+                long epoch = TransactionalContext.getCurrentContext().hintsEpoch;
+                return new TokenResponse(ts, epoch, null);
+            } else {
+                return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
+                        .nextToken(streamIDs, numTokens)));
+            }
+        } else {
+            return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
+                    .nextToken(streamIDs, numTokens)));
+        }
     }
 
-
     public TokenResponse nextToken(Set<UUID> streamIDs, int numTokens,
-                                   TxResolutionInfo conflictInfo) {
+                                  TxResolutionInfo conflictInfo) {
         return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
                 .nextToken(streamIDs, numTokens, conflictInfo)));
     }
 
+    private Long routeCall(Set<UUID> streamIDs) {
```

Comment:
can you please put comments on this method.I do not have the context.

Reply:
Done.